### PR TITLE
app,counter, runnable, thread classes implemented

### PR DESCRIPTION
**Pull request by semyonich**

### CODE REVIEW

File: src/main/java/com/multithread/RunnableClass.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        }
+        logger.info("Runnable finished!");
+    }
+
+    public void start() {
+        if (thread == null) {
+            thread = new Thread(this);
+            thread.start();
+        }
+    }
```
Comment (Ketch92) [inline src/main/java/com/multithread/RunnableClass.java]: it does seems you don't need to override this

File: src/main/java/com/multithread/Counter.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    }
+
+    public void setCounter(int counter) {
+        this.counter = counter;
+    }
+
+    public void increment() {
+        counter++;
+    }
+}
```
Comment (Ketch92) [inline src/main/java/com/multithread/Counter.java]: is it possible to solve it with a bit less amount of code?

**Review by Ketch92 — APPROVED** [review]

File: src/main/java/com/multithread/RunnableClass.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/com/multithread/RunnableClass.java]: its not overriding, in Runnable we have no start(), so we need to implement it

File: src/main/java/com/multithread/Counter.java (hunk 2 above)
Comment (semyonich) [reply src/main/java/com/multithread/Counter.java]: setter removed, we dont use it

File: src/main/java/com/multithread/Application.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.multithread;
+
+public class Application {
+    private static final int ITERATIONS_NUMBER = 100;
+    private static final int COUNTER_START = 0;
+
+    public static void main(String[] args) {
+        Counter counter = new Counter(COUNTER_START);
```
Comment (Sofasmile) [inline src/main/java/com/multithread/Application.java]:
```suggestion
        Counter counter = new Counter();
```
0 це дефолтне значення для числових типів

File: src/main/java/com/multithread/RunnableClass.java (hunk 1 above)
Comment (Sofasmile) [inline src/main/java/com/multithread/RunnableClass.java]: це краще винести в main метод

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/multithread/Application.java (hunk 3 above)
Comment (semyonich) [reply src/main/java/com/multithread/Application.java]: done

File: src/main/java/com/multithread/RunnableClass.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/com/multithread/RunnableClass.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**